### ISSUE TITLE
Cannot reduce marks when marks are not present

### ISSUE THREAD

**Issue by Aaduriz** [contentful/rich-text#111]
I am retrieving a `Rich Text` field from my Contentful model and since I have not introduced any types into it, the GraphQL blows up when I try and retrieve them (because there are none).
If I remove the `marks { type }` JSON blob part, the GraphQL executes successfully, but the `rich-text-react-renderer` is expecting to have an existing object called `marks` to apply some styling.
My surprise is that there's no other way for me to create React components out of a JSON blob without marks. 
I would expect that if part of the document that I am passing to the `rich-text-react-renderer` component doesn't contain marks, for it not to blow up and show the content without any marks...

**Comment by Aaduriz** [comment]
![image](https://user-images.githubusercontent.com/40381836/62542088-635bb500-b85b-11e9-89b4-99a76ed83019.png)

**Comment by cribbles** [comment]
Hey @Aaduriz,

>My surprise is that there's no other way for me to create React components out of a JSON blob without marks.

For rich text markup, the `marks` property is expected to present on `text` nodes even when there are no marks applied to the text (in that case, as an empty array).

Could you explain more about your motive behind removing the `marks` property from the JSON?

**Comment by Aaduriz** [comment]
When I declare the GraphQL query, since there are no marks in any of the different node types, if I declare that I want to retrieve marks, the GraphQL query throws an exception.

**Comment by cribbles** [comment]
Hi @Aaduriz 

Are you using the Contentful GraphQL API or an external library? We've seen similar problems in the past with, e.g., Gatsby's GraphQL plugin, which generates schemas programmatically and throws an error when it encounters an unknown field (in this case, `marks`).

If that's the case, could you try implementing a resolver to add a `marks: []` property to `text` nodes returned from the GraphQL query result? Unfortunately, we can't change the `rich-text-react-renderer` itself for this use-case - the `marks` property is required for `text` nodes, so even if we patched this area of the library for cautious navigation, a payload without the `marks` object would cause similar errors elsewhere.

**Comment by Aaduriz** [comment]
> Are you using the Contentful GraphQL API or an external library? We've seen similar problems in the past with, e.g., Gatsby's GraphQL plugin, which generates schemas programmatically and throws an error when it encounters an unknown field (in this case, marks).

I am using Gatsby to query for Contentful content, yes

> If that's the case, could you try implementing a resolver to add a marks: [] property to text nodes returned from the GraphQL query result?
I was kinda expecting that the package would be able to disregard the marks if they are not present. If that's not the case, I shall have to implement a resolver as you mention for most of the types, and if there are no marks, to add an empty array.

**Comment by rmcsharry** [comment]
@Aaduriz I just stumbled into this also with GatsbyJS. It seems it cannot see the "Marks" property as @cribbles mentioned.

Did you manage to write a resolver? I have no clue about that, new to React, Gatsby and Contentful.

Is there maybe a guide on how to implement a resolver to fix this?

**Comment by rmcsharry** [comment]
@cribbles I think actually this is a bug in the Contentful editor for Rich Text.

See screenshot.

I created two nodes, both of type Rich Text.

1. subTitle - I simply typed and did not use any of the format tools.
2. textA - I added bold to the text and made it heading-3.

When I query them, the subTitle has no marks array (I would expect an empty array since the user did not use any marks, only typed text).

![image](https://user-images.githubusercontent.com/1156100/65395615-c5399500-dd9d-11e9-8a32-2594c7e30bea.png)

**Comment by fasaas** [comment]
@rmcsharry What I ended up doing, but I am still not proud of, is to create a `default` entry of the same type with the minimal set of content to avoid having this error. This is making me to be tightly coupled to specific content entries for fallback values, which I expected I never had to be dependent on.

**Comment by rmcsharry** [comment]
@fasaas Thanks, interesting solution, but I don't want to be tightly coupled either. I think it's a but in the Contentful Rich Text editor. I can work around it for now by just making sure all fields have some formatting so marks get created.

**Comment by rmcsharry** [comment]
@cribbles you said:

> For rich text markup, the marks property is expected to present on text nodes even when there are no marks applied to the text 

But that is NOT happening.

For example, create a RTN and just type some text. There is no marks property created. Then go back and make the text italic or bold. Now there is a marks property.

So it seems the user has to do some kind of formatting otherwise there is no marks property. Is this because Contentful is not creating the marks property or is it because the GraphQL cannot see it if no formatting was applied?

**Comment by vpelkonen** [comment]
I was able to circumvent the same error by taking only the JSON data through GraphQL:
```
someRichTextNode {
  json
}
```
...and passing the JSON through to `documentToReactComponents`:
```
documentToReactComponents(someRichTextNode.json)
```
Source: https://www.contentful.com/developers/docs/tutorials/general/rich-text-and-gatsby/

**Comment by rmcsharry** [comment]
@vpelkonen Yep that's what I ended up doing also. I still think the root problem is a bug in Contentful though since I can repeatedly make it happen as described by adding/ removing formatting. @cribbles ?
